keyboards/monsgeek/m2/keyboard.json

```diff
@@ -0,0 +1,278 @@
+{
+    "manufacturer": "MonsGeek",
+    "keyboard_name": "M2",
+    "maintainer": "jonylee@hfd",
+    "bootloader": "wb32-dfu",
+    "diode_direction": "ROW2COL",
+    "eeprom": {
+        "driver": "wear_leveling",
+        "wear_leveling": {
+            "driver": "spi_flash",
+            "backing_size": 8192
+        }
+    },
+    "dynamic_keymap": {
+        "layer_count": 6
+    },
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": false,
+        "nkro": true,
+        "rgb_matrix": true
+    },
+    "indicators": {
+        "caps_lock": "A15",
+        "num_lock": "C10"
+    },
+    "qmk": {
+        "locking": {
+            "enabled": true,
+            "resync": true
+        }
+    },
+    "matrix_pins": {
+        "cols": ["C1", "C2", "C3", "A0", "A1", "A2", "A3", "A4", "A5", "A6", "A7", "C4", "C5", "B0", "B1", "B2", "B10", "B11", "B12", "B13", "B14"],
```

Comment:
Col 14, pin `B1` is never used on this board.  Why map it?  Just adds a moot step to the matrix loop, at each scan.
*(That col 15 and 16 are only used in the arrows is inefficient, but that's what it is and can't be changed* :( *.)*